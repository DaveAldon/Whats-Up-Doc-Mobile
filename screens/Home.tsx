import * as React from "react";
import { StyleSheet, Image, ScrollView, TouchableOpacity } from "react-native";
import * as Diseases from "../constants/Diseases";
import { Text, View } from "../components/Themed";
import { useTheme } from "@react-navigation/native";
import * as enums from "../constants/enums";
import { AntDesign } from "@expo/vector-icons";
import { useEffect } from "react";
import * as Device from "expo-device";
import ErrorBoundary from "../components/ErrorBoundary";

interface IDisease {
  name: string;
  code: string;
}

export default function Home(props: any) {
  const navImg = require("../assets/images/icon.png");
  const { colors } = useTheme();
  const { navigation } = props;

  // Set a custom header
  useEffect(() => {
    navigation.setOptions({
      headerTitle: () => (
        <View
          style={{
            flexDirection: "row",
            width: "100%",
            backgroundColor: enums.colors.transparent,
            justifyContent: "center",
            alignItems: "center",
            paddingBottom: 10,
          }}
        >
          <View
            style={{ flexDirection: "row", justifyContent: "center", width: 40, backgroundColor: enums.colors.transparent, height: 40, margin: 5 }}
          >
            <Image source={navImg} style={{ height: 40, width: 40, resizeMode: "cover", borderRadius: 10 }} />
          </View>
          <View style={{ backgroundColor: enums.colors.transparent, flexDirection: "row", justifyContent: "center", margin: 5 }}>
            <Text style={{ fontSize: 18, fontWeight: "500", color: colors.text }}>What's Up Doc</Text>
          </View>
        </View>
      ),
      headerBackTitle: " ",
      headerStyle: {
        backgroundColor: colors.background,
      },
    });
  });

  return (
    <View style={styles.container}>
      <Text style={{ padding: 10 }}>
        Tap on a disease to search for providers that specialize in that area, according to the provider registry you can provide in the Settings. The
        default registry is HL7 FHIR's sample database.
      </Text>
      <ScrollView style={{ width: Device.modelName === "iPad" ? "50%" : "100%", padding: 10 }}>
        {Diseases.DiseaseMapping.map((disease: IDisease, index: number) => (
          <TouchableOpacity
            key={index}
            onPress={async () => {
              navigation.navigate(enums.SCREENS.LISTING, {
                code: disease.code,
                disease: disease.name,
              });
            }}
          >
            <View style={[{ backgroundColor: colors.border }, styles.card]}>
              <Text style={{ color: colors.text }}>{disease.name}</Text>
              <ErrorBoundary>
                <AntDesign name="arrowright" size={24} color={colors.text} />
              </ErrorBoundary>
            </View>
          </TouchableOpacity>
        ))}
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  card: {
    height: 50,
    marginBottom: 10,
    borderRadius: 10,
    flexDirection: "row",
    alignItems: "center",
    paddingHorizontal: 20,
    justifyContent: "space-between",
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
});
