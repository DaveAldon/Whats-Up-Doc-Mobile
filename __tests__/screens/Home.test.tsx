import React from "react";
import renderer from "react-test-renderer";
import Home from "../../screens/Home";

const mockProp = {
  navigation: {
    setOptions: jest.fn(),
  },
};

jest.mock("react-native-gesture-handler", () => {});

describe("<Home />", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("has children", () => {
    const tree = renderer.create(<Home {...mockProp} />).toJSON();
    expect(tree?.children.length).toBeGreaterThan(0);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Home {...mockProp} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
