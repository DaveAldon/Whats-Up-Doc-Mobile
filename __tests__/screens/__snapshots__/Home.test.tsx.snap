// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "padding": 10,
        },
      ]
    }
  >
    Tap on a disease to search for providers that specialize in that area, according to the provider registry you can provide in the Settings. The default registry is HL7 FHIR's sample database.
  </Text>
  <RCTScrollView
    style={
      Object {
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Array [
                Object {
                  "backgroundColor": "rgb(216, 216, 216)",
                },
                Object {
                  "alignItems": "center",
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "space-between",
                  "marginBottom": 10,
                  "paddingHorizontal": 20,
                },
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                },
              ]
            }
          >
            Coronary Artery Disease
          </Text>
          <Text />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Array [
                Object {
                  "backgroundColor": "rgb(216, 216, 216)",
                },
                Object {
                  "alignItems": "center",
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "space-between",
                  "marginBottom": 10,
                  "paddingHorizontal": 20,
                },
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                },
              ]
            }
          >
            Hypertension
          </Text>
          <Text />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Array [
                Object {
                  "backgroundColor": "rgb(216, 216, 216)",
                },
                Object {
                  "alignItems": "center",
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "space-between",
                  "marginBottom": 10,
                  "paddingHorizontal": 20,
                },
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                },
              ]
            }
          >
            Lyme Disease
          </Text>
          <Text />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Array [
                Object {
                  "backgroundColor": "rgb(216, 216, 216)",
                },
                Object {
                  "alignItems": "center",
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "space-between",
                  "marginBottom": 10,
                  "paddingHorizontal": 20,
                },
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                },
              ]
            }
          >
            Measles
          </Text>
          <Text />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Array [
                Object {
                  "backgroundColor": "rgb(216, 216, 216)",
                },
                Object {
                  "alignItems": "center",
                  "borderRadius": 10,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "space-between",
                  "marginBottom": 10,
                  "paddingHorizontal": 20,
                },
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                },
              ]
            }
          >
            Bronchitis
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
